### DISCUSSION TITLE
customize enhance plugin not works

### DISCUSSION THREAD

**Discussion by mgyboom** [apache/skywalking#10759] [Q&A] [answered]
I expect that method 'case1' could be enhanced.
```java
package cn.tongdun.demoskywalking.service.impl;

import com.alibaba.fastjson2.JSON;
import cn.tongdun.demoskywalking.domain.Case1DTO;
import cn.tongdun.demoskywalking.service.DemoService;
import cn.tongdun.ionicclient.IonicProducer;
import cn.tongdun.ionicclient.Msg;
import lombok.extern.slf4j.Slf4j;
import org.apache.skywalking.apm.toolkit.trace.ActiveSpan;
import org.apache.skywalking.apm.toolkit.trace.Tag;
import org.apache.skywalking.apm.toolkit.trace.Trace;
import org.apache.skywalking.apm.toolkit.trace.TraceContext;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.scheduling.annotation.Async;
import org.springframework.stereotype.Service;

import javax.annotation.Resource;
import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

@Slf4j
@Service
public class DemoServiceImpl implements DemoService {

    @Value("${topic}")
    private String topic;

    @Resource
    private IonicProducer ionicProducer;

//    @Trace
//    @Tag(key = "k1", value = "arg[0]")
//    @Tag(key = "k2", value = "returnedObj.key2")
    @Override
    public Case1DTO case1(String param0) {
//        ActiveSpan.setOperationName("op:case1");
//        TraceContext.putCorrelation("k", "v");
        log.info("DemoService Case 1, {}", param0);
        final Case1DTO case1DTO = new Case1DTO();
        case1DTO.setKey1("a");
        case1DTO.setKey2("b");
        case1DTO.setKey3("c");
        return case1DTO;
    }
}
```
customize_enhance.xml as follows:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<enhanced>
    <class class_name="cn.tongdun.demoskywalking.service.impl.DemoServiceImpl">
        <method method="case1(java.lang.String)" operation_name="/case1" static="false">
            <tag key="k1">arg[0]</tag>
        </method>
    </class>
</enhanced>
```
I have configured the customize_enhance.xml file as follows:
![image](https://user-images.githubusercontent.com/45807330/236161311-f1607d81-7ff4-478e-b503-23a6031ba8b8.png)
I moved optional-plugins/apm-customize-enhance-plugin.jar to plugin/apm-customize-enhance-plugin.jar.
And I added `plugin.customize.enhance_file=C:\Users\mgy\skywalking-agent\config\customize_enhance.xml` into agent.config.

The javaagent command is as follows:
`-javaagent:C:\Users\mgy\skywalking-agent\skywalking-agent.jar=agent.service_name=demo-skywalking-1,collector.backend_service=10.58.14.24:21800`

But It does not work as expected.I can not find tag in this trace.
![image](https://user-images.githubusercontent.com/45807330/236163370-56280e06-4ec4-41c1-bf61-e10d169fe11d.png)

**Comment by a2a94** [comment]
try to change the window path separator with "\\" in agent.conf

**Reply by mgyboom** [reply]
> try to change the window path separator with "\" in agent.conf

It is already a path split with "\\"，as follows:
`plugin.customize.enhance_file=C:\Users\mgy\skywalking-agent\config\customize_enhance.xml`

**Reply by a2a94** [reply]
![image](https://user-images.githubusercontent.com/29352461/236606316-3d78dcfc-fb41-479e-a3c6-83c01a253489.png)
sorry，I just make a mistake.It should be Double slash.Otherwise, the path will be solve as Usersmgyskywalking-agentconfigcustomize_enhance.xml

**Reply by mgyboom** [reply, marked as answer]
> ![image](https://user-images.githubusercontent.com/29352461/236606316-3d78dcfc-fb41-479e-a3c6-83c01a253489.png) sorry，I just make a mistake.It should be Double slash.Otherwise, the path will be solve as Usersmgyskywalking-agentconfigcustomize_enhance.xml

It worked. Thanks for your answer.

**Comment by a2a94** [comment]
And I have the same question too. The plugin doesnt work at controller level but work  at a service or a static method.

**Reply by mgyboom** [reply]
> And I have the same question too. The plugin doesnt work at controller level but work at a service or a static method.

Indeed it is.
